### ISSUE TITLE
Prototype a System Two sibling to Jev over one shared cognition state

### ISSUE THREAD

**Issue by rinsakamo** [rinsakamo/relay-self#267]
## Summary

Prototype the smallest **System Two counterpart to Jev/System One** on the current Jev-capable llama.cpp substrate while preserving one RelayEngine owner and one cognition-state source.

Target execution shape:

~~~text
same cognition state / same bounded task
        |
        +-- System One -> LABEL_LOGITS
        |
        +-- System Two -> COMPLETION
~~~

This Issue is an implementation/design ledger for the execution substrate. It is not semantic authority and does not redefine RelaySelf cognition ownership.

Refs #259.
Refs #264.
Refs #266.
Follows #262.

## Fresh starting evidence

RelaySelf at issue creation:

~~~text
main = 6e9a302f6796479e319a35a5d1c6837cceb95edd
tree = c11550360eb5cb65c6fff05096ec12e2482e69ac
open PRs targeting main = 0
ruleset 23442682 = active
~~~

Current external Jev-capable substrate inspected fresh:

~~~text
repository = kishida/llama.cpp
branch = jev
HEAD = b779b6ac2ca517e83de42010e54fbe67acf1bb3c
~~~

These hashes are evidence only. Re-read current authority before execution.

## Current relevant llama.cpp facts

Current Jev branch already has two task families needed for this architecture:

~~~text
SERVER_TASK_TYPE_LABEL_LOGITS
SERVER_TASK_TYPE_COMPLETION
~~~

LABEL_LOGITS is explicitly the Jev /v1/systemone task family.

Current /v1/systemone:

1. accepts structured state and typed questions;
2. renders one user message;
3. applies the model's own chat template with thinking disabled;
4. evaluates the prompt;
5. reads candidate label logits at the last prompt position;
6. generates zero output tokens.

Normal chat/completion tasks already use SERVER_TASK_TYPE_COMPLETION.

Therefore a System Two sibling does not require a second model engine.

## Current Jev prompt shape

Current Jev user message is:

~~~text
Context:
<state>

Answer the question with only the label of the best option ...
Question: <instructions>
Options:
<label>: <option>
...
~~~

jev_prompt wraps that single user message through the model's own chat template with:

~~~text
add_generation_prompt = true
enable_thinking = false
~~~

plus an optional detected assistant prefix where required by the template.

This yields a useful shared state prefix, but the current System-One-specific instruction appears before Question/Options.

Therefore current S1/S2 token-prefix alignment can safely share:

~~~text
chat-template user header
Context:
<state>
~~~

but cannot automatically share the full Question/Options region if System Two uses different mode instructions.

## Phase 1 target — preserve qualified System One

Do not alter the current Jev/System One prompt first.

Prototype the System Two counterpart around the current prompt contract while preserving current #257/#262 evidence.

The first acceptable shape is approximately:

~~~text
same canonical state
same bounded semantic question / criteria

S1:
  existing /v1/systemone
  -> LABEL_LOGITS

S2:
  sibling generated path
  -> COMPLETION
  -> same admissible RelaySelf choice set
~~~

RelaySelf must continue to own the bounded answer-space contract.

System Two may generate rationale or other transient working text, but the final accepted result must still resolve only to:

~~~text
one request-carried choice_id
or
UNRESOLVED
~~~

No generated new option may silently widen the CLOSED topology.

## Endpoint question

A dedicated /v1/systemtwo endpoint is a candidate, not a prerequisite.

First compare the smallest two realizations.

### A. Adapter-only sibling

Render the same canonical cognition state into ordinary /v1/chat/completions using a shared prompt-framing function in the RelaySelf target adapter.

### B. Native sibling endpoint

Add a Jev-adjacent /v1/systemtwo path in the fork that reuses the same cognition request parser / prompt renderer and dispatches a SERVER_TASK_TYPE_COMPLETION.

Prefer B only if it materially reduces token-shape drift, duplicate parsing, or cache-prefix ambiguity.

Do not add an endpoint merely for naming symmetry.

## Shared renderer requirement

The architectural target is one canonical rendering seam for shared cognition material:

~~~text
render_common_cognition(...)
  -> stable/common token prefix

then

System One postfix
  -> label-logit read

System Two postfix
  -> generated bounded judgment
~~~

Do not rely on two independently maintained prompt templates accidentally remaining token-identical.

## Phase 2 candidate — maximize full task-prefix reuse

Only after Phase 1 preserves current System One behavior, evaluate whether the current Jev wording leaves materially useful prefix reuse on the table.

The stronger future layout would be:

~~~text
Context:
<state>

Question: ...
Options:
...

<mode-specific postfix>
~~~

with:

~~~text
S1 postfix:
  answer only with the best label

S2 postfix:
  perform deeper bounded cognition and return an admissible result
~~~

This would allow the common prefix to extend through the task itself.

However, changing current Jev prompt ordering invalidates direct comparability with existing Jev qualification.

Therefore:

~~~text
change System One prompt
  -> requires explicit requalification
~~~

Do not silently mutate it during Phase 1.

## Cache boundary

This Issue does not qualify or enable S1 -> S2 KV reuse.

Preserve:

~~~text
semantic shared state
!= token-prefix identity
!= physical KV reuse
~~~

#259 owns physical Jev -> THINK/System-Two cache qualification.

RelayLM #3006 remains the correctness dependency before repaired cache reuse is trusted.

All Phase 1 requests should keep cache reuse disabled unless current authority changes explicitly.

## Relation to #264

#264 owns SOUL / identity projection into a cache-stable cognition prefix.

This Issue should consume, not duplicate, that work.

Desired eventual ordering remains approximately:

~~~text
stable cognition material
  - common provider framing
  - SOUL / identity projection

decision-epoch material
  - Intent / Focus
  - Present
  - selected Memory
  - task / choices

execution discriminator
  - S1 or S2 postfix
~~~

No new semantic owner follows from the physical prompt layout.

## Deterministic acceptance criteria

Before any live cache qualification:

- [ ] System One and System Two consume the same semantic bounded request;
- [ ] the admissible choice_id set is identical across S1/S2;
- [ ] System Two cannot widen CLOSED topology with a generated new option;
- [ ] one shared cognition renderer owns the common state shape;
- [ ] current Jev/System One request shape remains unchanged in Phase 1;
- [ ] no hidden threshold/fallback is introduced;
- [ ] explicit UNRESOLVED remains representable;
- [ ] one RelayEngine provider owner remains;
- [ ] no cache reuse is enabled or claimed;
- [ ] tests freeze exact common-prefix rendering at the text/token-input boundary where practical.

## Physical acceptance later

After the deterministic shape exists and #259's correctness dependencies permit physical work, measure:

~~~text
S1 resolved
S1 unresolved -> S2 cold
S1 unresolved -> S2 reuse candidate
~~~

and require separate evidence for semantic equivalence, exact prompt-prefix identity, observed cache reuse, reduced processed prompt work, valid stop termination, same admissible outcome, and no retry/replay/fallback contamination.

Latency alone is insufficient.

## Grand Null

Do not add:

- a second RelayEngine;
- a SystemTwo semantic owner;
- a second model requirement;
- hidden automatic escalation inside llama.cpp;
- generated answer-space expansion;
- a generic cognition router;
- cache enablement;
- a new persistent state store;
- a new conversation owner;
- a Jev prompt rewrite merely for aesthetics.

## Completion

Phase 1 completes when the repository has a deterministic, testable System Two sibling shape over the same bounded semantic request, with current System One preserved and cache reuse still disabled.

Any physical S1 -> S2 reuse claim remains governed by #259.

The design question is:

> Can one Jev-capable llama.cpp runtime execute the same cognition state as either a zero-generation typed read or a generated deeper bounded judgment without creating a second engine or changing semantic ownership?

**Comment by rinsakamo** [comment]
## Terminal reconciliation — Phase 1 deterministic sibling shape completed

PR #270 merged as:

~~~text
main = 6c51a3d2d7f45a32637e116e301262905f482b5c
tree = 2fa9291e8bba3211db61bed4d2f2f7a4ba0855cd
~~~

Exact PR head `e58dcd3d90b2b78aeb8508bed2000bc1f9a656b2` passed:

- repository-contracts
- pytest
- lint
- mineflayer-adapter

The merged deterministic shape now establishes:

~~~text
same BoundedChoiceRequest
        |
        +-- shared bounded cognition state renderer
        |
        +-- current Jev/System One request
        |      -> unchanged request body
        |
        +-- inactive System Two candidate renderer
               -> one-user-message generation shape
               -> Context:\n<same canonical state>\n\n...
               -> same request-carried choice_id set
               -> explicit unresolved remains available
~~~

Important boundaries preserved:

- the active provider path is unchanged;
- current Jev -> existing THINK escalation remains the runtime path;
- current Jev/System One request body is frozen unchanged by regression;
- System Two cannot widen the request-carried CLOSED answer space through its response schema;
- cache_prompt remains false;
- no cache/KV reuse claim is made;
- no /v1/systemtwo endpoint was added;
- no second RelayEngine, model, semantic owner, or persistent state was added.

This completes the deterministic Phase 1 scope of this Issue.

The remaining physical question is independent and already owned by #259:

~~~text
semantic shared state
  != exact token prefix
  != observed KV reuse
~~~

In particular, Python's candidate state rendering is not by itself evidence that the Jev C++ chat-template path produces an exact reusable token prefix. #259 must establish that physically after its RelayLM correctness dependency permits it.

No further repository mutation is required under #267.
